docs/extra.rst

```diff
@@ -0,0 +1,116 @@
+Extra
+=====
+
+Chained select2
+---------------
+
+Suppose you have an address form where a user should choose a Country and a City.
+When the user selects a country we want to show only cities belonging to that country.
+So the one selector depends on another one.
+
+Models
+``````
+
+Here are our two models:
+
+.. code-block:: python
+
+    class Country(models.Model):
+        name = models.CharField(max_length=255)
+
+
+    class City(models.Model):
+        name = models.CharField(max_length=255)
+        country = models.ForeignKey('Country', related_name="cities")
+
+
+Customizing a Form
+``````````````````
+
+Lets link two widgets via *dependent_fields*.
+
+.. code-block:: python
+    :emphasize-lines: 15
+
+    class AddressForm(forms.Form):
+        country = forms.ModelChoiceField(
+            queryset=Country.objects.all(),
+            label=u"Country",
+            widget=ModelSelect2Widget(
+                search_fields=['name__icontains'],
+            )
+        )
+
+        city = forms.ModelChoiceField(
+            queryset=City.objects.all(),
+            label=u"City",
+            widget=ModelSelect2Widget(
+                search_fields=['name__icontains'],
+                dependent_fields=['country'],
+                max_results=500,
+            )
+        )
+
+
+Interdependent select2
```

Comment:
Why are you so sure?
Possibly it has some strange behaviour for parent-child relation, when child is selected you cannot change parent, but it works.